src/main/java/com/puppycrawl/tools/checkstyle/api/AstFilter.java

```diff
@@ -0,0 +1,34 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.api;
+
+/**
+ * An interface for filtering {@code TreeWalkerAuditEvent}.
+ *
+ * @author Timur Tibeyev.
+ */
+public interface AstFilter {
```

Comment:
@timurt , @MEZk 
I do not remember that I allowed new classes in API.
We discussed to keep then out of API till we finish implementation, to make sure that tested that contract and we could keep it.
One more point is names of interfaces: AstFilter vs TreeWalkerAuditEvent .
it is better to name filter as TreeWalkerFilter.